components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -109,4 +116,12 @@ export class ProjectsFilterDropdownComponent {
       (element as HTMLElement).focus();
     }
   }
+
+  checkInitialEquality(): void {
+    // Check for deep equality against new changes, disabling
+    // apply button if changes are not new
+    JSON.stringify(this.initialFilters) === JSON.stringify(this.editableOptions)
```

Comment:
This is an expensive comparison, likely too expensive, given that this method will likely be called every digest cycle. Two factors contribute to the expense: (a) converting each object to a string, and (b) including all properties of each `ProjectsFilterOption` when we only care about the `checked` property.

Two alternatives to consider.

**(1) Tighten the data.** 

First, rather than a copy of the entire array, take a "slice" of just the `checked` property:
```
this.initialFilters = this.filteredOptions.map(o => o.checked);
```
Then use the lodash/fp `isEqual` operator rather than `stringify` to be more efficient.

**(2) Track the changes as they occur.**

This is more in keeping with the Angular style but is just a tad more work to set up. That is, starting with `this.initialFilters` being a boolean array initialized to all false, when the user checks or unchecks an item, toggle the corresponding array element of `this.initialFilters`, too. Then the `disabled` property of the button would just need to watch `!this.initialFilters.some(identity)`, i.e., the Apply button should be enabled if-and-only-if any element of `this.initialFilters` is true.

Reply:
Thanks for the suggestion.  I ended up going with option one because it was easier to follow.

Ultimately i'm using `isEqual`, instead of `isEqualWith` and a customizer function because the I couldn't quite get the customizer to operate in the fashion I expected it to.